lnbits/wallets/cln.py

```diff
@@ -146,34 +145,34 @@ async def pay_invoice(self, bolt11: str, fee_limit_msat: int) -> PaymentResponse
             True, r["payment_hash"], fee_msat, r["payment_preimage"], None
         )
 
-    async def get_invoice_status(self, checking_id: str) -> PaymentStatus:
+    async def get_invoice_status(self, payment: Payment) -> PaymentStatus:
         try:
-            r = self.ln.listinvoices(payment_hash=checking_id)
+            r: dict = self.ln.listinvoices(payment_hash=checking_id)  # type: ignore
         except:
             return PaymentStatus(None)
         if not r["invoices"]:
             return PaymentStatus(None)
 
         invoice_resp = r["invoices"][-1]
 
-        if invoice_resp["payment_hash"] == checking_id:
+        if invoice_resp["payment_hash"] == payment.checking_id:
             if invoice_resp["status"] == "paid":
                 return PaymentStatus(True)
             elif invoice_resp["status"] == "unpaid":
                 return PaymentStatus(None)
-        logger.warning(f"supplied an invalid checking_id: {checking_id}")
+        logger.warning(f"supplied an invalid checking_id: {payment.checking_id}")
         return PaymentStatus(None)
 
-    async def get_payment_status(self, checking_id: str) -> PaymentStatus:
+    async def get_payment_status(self, payment: Payment) -> PaymentStatus:
         try:
-            r = self.ln.call("listpays", {"payment_hash": checking_id})
+            r: dict = self.ln.call("listpays", {"payment_hash": checking_id})  # type: ignore
         except:
             return PaymentStatus(None)
```

Comment:
the `checking_id` variable here doesnt exist which causes the tests to fail. (needs to be `payment.checking_id`
doesnt get caught because of the `# type: ignore` though

the broad `except` doesnt really help either as it swallows the error